src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,64 @@
+package mate.academy.dao.impl;
+
+import jakarta.persistence.Query;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Unable to add movieSession", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Unable to retrieve movieSession", e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery(
+                    "from MovieSession ms "
+                            + "left join fetch ms.movie " + "left join fetch ms.cinemaHall "
+                            + "where ms.movie.id = :movieId " + "and ms.showTime >= :startOfDay "
+                            + "and ms.showTime < :endOfDay", MovieSession.class);
+            query.setParameter("movieId", movieId);
```

Comment:
Can a movie session exist without a movie or cinema hall? Let’s  use the correct type of join